### PR TITLE
Added settings field for comment sub text

### LINKED ISSUE #1502: Comments: Allow subscription text to be edited
In some cases users may want to change the default "Notify me of new comments via email." From what I can find, that phrase is hard-coded in the subscriptions module, probably for ease of translation.

### PR DESCRIPTION
**Pull request by chaselivingston**
Fixes #1502 in progress

### CODE REVIEW

**Comment by samhotchkiss** [comment]
@chaselivingston -- this patch looks pretty good, but I would propose that this should be able to be overridden with a filter rather than through the admin.  I think we should be working to remove options, especially when they will (likely) be used by very few people.

Thoughts?

**Comment by jeherve** [comment]
> I would propose that this should be able to be overridden with a filter rather than through the admin

:+1:

**Comment by dereksmart** [comment]
+1, adding options for text to be overridden is a slippery slope.  If we're allowing this text to be changed, why not "Notify me of new posts via email."?

**Comment by jeherve** [comment]
> If we're allowing this text to be changed, why not "Notify me of new posts via email."?

It would actually be nice to create 2 filters, one for each setting, imo.

**Comment by samhotchkiss** [comment]
@jeherve: Agreed

@chaselivingston: would you mind reworking this?  Your code looks good, but optionitis!  :)

**Comment by chaselivingston** [comment]
Yep, I'll work on adding filters for both of those instead :)

**Comment by chaselivingston** [comment]
@samhotchkiss @jeherve Here's the new PR with filters instead, what do you think?

File: modules/comments/admin.php [hunk 1]
```diff
@@ -14,6 +14,12 @@ class Jetpack_Comments_Settings {
 	 * @var string
 	 */
 	var $default_greeting = ''; // Set in constructor
+	
+	/**
+	 * The default comment form subscription text
+	 * @var string
+	 */
+	var $default_comment_subscription = ''; // Set in constructor
```
Comment (samhotchkiss) [inline modules/comments/admin.php]: We can drop this

File: modules/comments/admin.php [hunk 2]
```diff
@@ -48,6 +54,7 @@ public function __construct( Highlander_Comments_Base $jetpack_comments ) {
 	protected function setup_globals() {
 		// Default option values
 		$this->default_greeting = __( 'Leave a Reply', 'jetpack' );
+		$this->default_comment_subscription = __( 'Notify me of new comments via email.', 'jetpack' );
```
Comment (samhotchkiss) [inline modules/comments/admin.php]: and this

File: modules/comments/admin.php [hunk 3]
```diff
@@ -155,6 +164,14 @@ function comment_form_greeting_sanitize( $val ) {
 	}
 
 	/**
+	 * Sanitize the comment subscription text
+	 *
+	 * 
+	 * @param type $val
+	 * @return string
+	 */
```
Comment (samhotchkiss) [inline modules/comments/admin.php]: and this docbloc

File: modules/subscriptions.php [hunk 4]
```diff
@@ -471,14 +471,16 @@ function comment_subscribe_init() {
 		if ( FALSE === has_filter( 'comment_form', 'show_subscription_checkbox' ) && 1 == get_option( 'stc_enabled', 1 ) && empty( $post->post_password ) ) {
 			// Subscribe to comments checkbox
 			$str .= '<p class="comment-subscription-form"><input type="checkbox" name="subscribe_comments" id="subscribe_comments" value="subscribe" style="width: auto; -moz-appearance: checkbox; -webkit-appearance: checkbox;"' . $comments_checked . ' /> ';
-			$str .= '<label class="subscribe-label" id="subscribe-label" for="subscribe_comments">' . __( 'Notify me of follow-up comments by email.', 'jetpack' ) . '</label>';
+			$comment_sub_text = __( 'Notify me of follow-up comments by email.', 'jetpack' );
+			$str .=	'<label class="subscribe-label" id="subscribe-blog-label" for="subscribe_blog">' . esc_html( apply_filters( 'jetpack_subscribe_comment_label', $comment_sub_text ) ) . '</label>';
```
Comment (samhotchkiss) [inline modules/subscriptions.php]: We should keep the "id" and "for" from the previous field

**Comment by samhotchkiss** [comment]
@chaselivingston just a couple super minor things, noted inline above

**Pull request merged**